src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1967 @@ public static void TestBlockingCollectionBasic()
+                });
+            }
+
+            //consumers
+            Task[] consumers = new Task[noOfConsumers];
+            for (int consumerIndex = 0; consumerIndex < noOfConsumers; consumerIndex++)
+            {
+                consumers[consumerIndex] = Task.Run(() =>
+                {
+                    while (!m_BlockingQueueUnderTest.IsCompleted)
+                    {
+                        long item;
+                        if (m_BlockingQueueUnderTest.TryTake(out item, 1))
+                        {
+                            int i = 0;
+                            for (int j = 0; j < 5; j++)
+                            {
+                                i += j;
+                            }
+                        }
+                    }
+                });
+            }
+
+            //Wait for the producers to finish.
+            //It is possible for some of the tasks in the array to be null, because the
+            //test was cancelled before all the tasks were creates, so we filter out the null values
+            Task.WaitAll(producers);
+
+            m_BlockingQueueUnderTest.CompleteAdding(); //signal all producers are done adding items
+
+            //Wait for the consumers to finish.
+            Task.WaitAll(consumers);
+
+            // success is not suffering exceptions.
+        }
+
+        /// <summary>
+        /// Verify the bug fix by making sure TryTakeFromAny succeeds and returns the correct index, the bug reason is
+        /// TryTakeFromAny construct a copy list of BCs and  another list for the wait handles, and calls WaitAny using these handles
+        /// If it fails (one or more of the collections completed) it removes the completed BC wait handles from the handles list and retry again
+        /// We don’t remove the correspondent BC from the list!
+        /// 
+        /// Now er update the BC list copy as well
+        /// </summary>
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest_Bug914998()
+        {
+            var producer1 = new BlockingCollection<int>();
+            var producer2 = new BlockingCollection<int>();
+            var producerArray = new BlockingCollection<int>[] { producer2, producer1 };
+
+            producer2.CompleteAdding();
+
+            Task.Factory.StartNew(() =>
+            {
+                producer1.Add(100);
+            }, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
+
+            int ignored, index, timeout = 5000;
+            index = BlockingCollection<int>.TryTakeFromAny(producerArray, out ignored, timeout);
+            if (index != 1)
+            {
+                Assert.False(true, String.Format("RunBlockingCollectionTest_Bug914998:  >Failed, TryTakeFromAny failed and returned {0} expected 1", index));
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTestConstruction()
```

Comment:
Why OuterLoop?  I'll stop asking, but we should review why it is we're making so many of these OuterLoop.